modules/lang-painless/src/main/java/org/elasticsearch/painless/CompilerSettings.java

```diff
@@ -137,4 +142,18 @@ public boolean areRegexesEnabled() {
     public void setRegexesEnabled(boolean regexesEnabled) {
         this.regexesEnabled = regexesEnabled;
     }
+
+    /**
+     * The whitelist to compile against.
+     */
+    public Definition getDefinition() {
+        return definition;
+    }
+
+    /**
+     * Set the whitelist to compile against.
+     */
+    public void setDefinition(Definition definition) {
```

Comment:
This *feels* like the more natural place to configure the whitelist, though I'm not sure about it.